src/infrastructure/noteAttachmentUploader.repository.ts

```diff
@@ -0,0 +1,63 @@
+import type NotesApiTransport from './transport/notes-api';
+import generateHash from './utils/hash';
+import getExtensionByMimetype from './utils/mimetypes';
+import type NoteAttachmentUploaderRepositoryInterface from '@/domain/noteAttachmentUploader.repository.interface';
+import type { Note } from '@/domain/entities/Note';
+
+/**
+ * Facade for interacting with files, which are note attachments
+ */
+export default class NoteAttachmentUploaderRepository implements NoteAttachmentUploaderRepositoryInterface {
+  /**
+   * Transport instance
+   */
+  private readonly transport: NotesApiTransport;
+
+  /**
+   * Repository constructor
+   * @param notesApiTransport - notes api transport instance
+   */
+  constructor(notesApiTransport: NotesApiTransport) {
+    this.transport = notesApiTransport;
+  }
+
+  /**
+   * Upload file data as note attachment
+   * @param noteId - indentifier for note to make attachment
+   * @param fileData - file binary data
+   * @returns key, by which we can load it from api
+   */
+  public async upload(noteId: Note['id'], fileData: Blob): Promise<string> {
+    const form = new FormData();
+
+    /**
+     * Generate filename for form data
+     */
+    const fileName = generateHash();
+    /**
+     * Get file mime type and define extension by it
+     */
+    const mimeType = fileData.type;
+    const fileExtension = getExtensionByMimetype(mimeType);
```

Comment:
why do we need this step? Why not just take the original extension?

Reply:
i think, it may be better to set extension by data type, instead of parsing it from name on another layer

Reply:
removed defining file extension, because it is useless